ui/src/pages/AllStories.tsx

```diff
@@ -21,111 +85,247 @@ function createData(
   is_visible: boolean,
   author_first_name: string,
   author_last_name: string,
-  author_country: string
-) {
+  author_country: string,
+  image_url: string,
+  video_url: string,
+  content: string
+): StoryView {
   const author_name = `${author_first_name} ${author_last_name}`;
   return {
-    id,
+    ID: id,
     title,
     current_city,
     year,
     author_name,
+    author_first_name,
+    author_last_name,
     author_country,
     is_visible,
+    image_url,
+    video_url,
+    content,
   };
 }
 
-const StyledAppBar = styled(AppBar)`
-  //styling
-`;
+interface State {
+  tabValue: number;
+  visibleState: StoryView[];
+  visibleTableState: StoryView[];
+  tableData: StoryView[];
+  changedVisibility: StoryView[];
+  selectedRowIds: number[];
+  order: "asc" | "desc";
+  orderBy: string;
+}
 
-export const AllStories: React.FC = () => {
-  const { data: allStories, error } = useSWR<Story[]>("/api/stories");
-  let rows = [];
+type Action =
+  | { type: "SWITCH_TAB"; id: number }
+  | { type: "HANDLE_SWITCH_CHANGE"; e: React.ChangeEvent; story: StoryView }
+  | { type: "INITIALIZE_AFTER_API"; rows: StoryView[] }
+  | { type: "HANDLE_CHECKED_ALL"; rows: StoryView[] }
+  | { type: "HANDLE_CHECKED"; e: React.ChangeEvent; story: StoryView }
+  | { type: "SET_ORDERING"; order: "asc" | "desc"; orderBy: string }
+  | { type: "SET_TABLE_DATA"; data: StoryView[] }
+  | { type: "SET_TAB_VALUE"; newValue: number };
 
-  // if (allStories===undefined) return;
-  if (allStories) {
-    rows = allStories.map((story, i) =>
-      createData(
-        i,
-        story.title,
-        story.current_city,
-        story.year,
-        story.is_visible,
-        story.author_first_name,
-        story.author_last_name,
-        story.author_country
-      )
-    );
+const initialState: State = {
+  tabValue: 0,
+  visibleState: [] as StoryView[],
+  visibleTableState: [] as StoryView[],
+  tableData: [] as StoryView[],
```

Comment:
We don't need to cast the types if we already set the record type above with `initialState: State`